api/v0/proto/retrievability/types.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package celestia.v0.retrievability.types;
+
+// TODO: oneof different blob versions
+message Blob {
+  // The blob data.
+  bytes data = 1;
+  // The signer of the blob.
+  optional bytes signer = 2;
+  // The blob commitment.
+  bytes commitment = 3;
+}
+
+// Blob with blob proof. Either the blob and the proof are optional, depending
+// on the options of the request.
+message BlobWithProof {
+  // The blob.
+  optional Blob blob = 1;
+  // The proof of blob inclusion or non-inclusion.
+  optional DataRootProof proof = 2;
+  // Index of the first share of the blob in the ODS.
+  uint64 index = 3;
+}
+
+// A single data root proof can span multiple rows.
+message DataRootProof {
+  // The type of the proof.
+  ProofType type = 1;
+  // The proof.
+  Proof proof = 2;
+}
+
+enum ProofType {
+  PROOF_TYPE_UNSPECIFIED = 0;
+  // Data inclusion proof.
+  PROOF_TYPE_DATA_INCLUSION = 1;
+  // Contains leaf hashes for the range where namespace would be.
+  PROOF_TYPE_DATA_NON_INCLUSION = 2;
+  // Commitment proof is used for blobs only.
+  // It contains first share of the blob and its inclusion proof.
+  // TODO: why is the first share included?
+  PROOF_TYPE_COMMITMENT = 3;
+  // TODO: structure for commitment non-inclusion.
+  PROOF_TYPE_COMMITMENT_NON_INCLUSION = 4;
```

Comment:
If I understand it correctly, this is for proving the inclusion of blobs right? Can we call this blob inclusion proofs (and perhaps share inclusion proofs). I just have an aversion to calling it commitment since the term is used everywhere

Reply:
Yeah blob inclusion/non-inclusion makes sense.

Reply:
The API also handles shares though, not just blobs.

Reply:
Simplified in https://github.com/celestiaorg/canonical-api/pull/2/commits/0daeb03f54c9e82ce8d9c90bdffb3d6ce6330f01 to actually just inclusion and non-inclusion. _What_ is being proved (blob, share, data?) doesn't have to be embedded in the proof itself.